rllib/core/rl_module/marl_module.py

```diff
@@ -273,6 +282,66 @@ def set_state(self, state_dict: Mapping[ModuleID, Any]) -> None:
         for module_id, state in state_dict.items():
             self._rl_modules[module_id].set_state(state)
 
+    def save_state_to_dir(self, dir: Union[str, pathlib.Path]) -> str:
+        """Saves the weights of this MultiAgentRLModule to dir.
+
+        Args:
+            dir: The directory to save the checkpoint to.
+
+        Returns:
+            The path to the saved checkpoint.
+        """
+        dir = pathlib.Path(dir)
+        dir.mkdir(parents=True, exist_ok=True)
+        for module_id, module in self._rl_modules.items():
+            module.save_to_checkpoint(str(dir / module_id))
+
+    def load_state_from_dir(
+        self, dir: Union[str, pathlib.Path], modules_to_load: Sequence
+    ) -> None:
+        """Loads the weights of an MultiAgentRLModule from dir.
+
+        Args:
+            dir: The directory to load the checkpoint from.
+            modules_to_load: The modules to load from the checkpoint. If None, all
+        """
+        dir = pathlib.Path(dir)
+        modules_to_load = set(modules_to_load)
+        if not modules_to_load:
+            modules_to_load = set(self._rl_modules.keys())
+        dir.mkdir(parents=True, exist_ok=True)
+        for module_id, module in self._rl_modules.items():
```

Comment:
why not loop over `modules_to_load` instead?

Reply:
done